### PR TITLE
LPS-110391 testing form instance published url on front-end

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-110391

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/util/FormURL.es.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+
+import FormURL from '../../../src/main/resources/META-INF/resources/admin/js/util/FormURL.es';
+
+const formInstanceId = 1;
+
+describe('FormURL', () => {
+	it('gets a blank url', () => {
+		const published = false;
+		const requireAuthentication = false;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/util/FormURL.es.js:22]: @marcosapmf, does it need to be `const` or we can use `let`, for example?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java [hunk 2]
```diff
@@ -186,6 +212,26 @@ protected DDMFormInstance mockDDMFormInstance(
 		return formInstance;
 	}
 
+	protected DDMFormInstance mockDDMFormInstance(
+			long formInstanceId, boolean published,
+			boolean requireAuthentication)
+		throws PortalException {
+
+		DDMFormInstance ddmFormInstance = mockDDMFormInstance(
+			formInstanceId, requireAuthentication);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java:221]: Good job reusing the existent method `mockDDMFormInstance`, @marcosapmf ! 👏

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java [hunk 3]
```diff
@@ -129,6 +130,31 @@ public void testGetFormURLForSharedFormInstance() throws Exception {
 		Assert.assertEquals(getSharedFormURL(), formURL);
 	}
 
+	@Test
+	public void testGetPublishedURLForPublishedForm() throws Exception {
+		DDMFormInstance publishedDDMFormInstance = mockDDMFormInstance(
+			_SHARED_FORM_INSTANCE_ID, true, false);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java:136]: @marcosapmf, can we add the boolean values to local variables as we did in the JavaScript tests?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java [hunk 4]
```diff
@@ -129,6 +130,31 @@ public void testGetFormURLForSharedFormInstance() throws Exception {
...
+				publishedDDMFormInstance);
+
+		Assert.assertEquals(
+			getSharedFormURL() + _SHARED_FORM_INSTANCE_ID, publishedFormURL);
+	}
+
+	@Test
+	public void testGetPublishedURLForUnpublishedForm() throws Exception {
+		DDMFormInstance unpublishedDDMFormInstance = mockDDMFormInstance(
+			_SHARED_FORM_INSTANCE_ID, false, false);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java:149]: @marcosapmf, can we add the boolean values to local variables as we did in the JavaScript tests?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/util/FormURL.es.js (hunk 1 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/util/FormURL.es.js:22]: @natocesarrego, it does not need to `const` but `formatSource` updates that

**Pull request closed without merging**